src/mahoji/lib/abstracted_commands/naxxusCommand.ts

```diff
@@ -0,0 +1,238 @@
+		totalPercent += rawPercent;
+		numKeys++;
+	}
+	totalPercent /= numKeys;
+	// Heavy penalize for having less than 50% in the main stat of this setup.
+	if (userStats[heavyPenalizeStat] < maxStats[heavyPenalizeStat] / 2) {
+		totalPercent = Math.floor(Math.max(0, totalPercent / 2));
+	}
+	return totalPercent;
+}
+
+export async function naxxusCommand(user: KlasaUser, channelID: bigint, quantity: number | undefined) {
+	const [hasReqs, rejectReason] = user.hasMonsterRequirements(Naxxus);
+	if (!hasReqs) {
+		return `${user.username} doesn't have the requirements for this monster: ${rejectReason}`;
+	}
+
+	const boosts = [];
+	let effectiveTime = Naxxus.timeToFinish;
+	if (isWeekend()) {
+		effectiveTime = reduceNumByPercent(effectiveTime, 5);
+		boosts.push('5% Weekend boost');
+	}
+
+	if ((user.settings.get(UserSettings.MonsterScores)[Naxxus.id] ?? 0) < 100) {
+		effectiveTime = increaseNumByPercent(effectiveTime, 15);
+		boosts.push('-15% <100 KC');
+	}
+
+	if ((user.settings.get(UserSettings.MonsterScores)[Naxxus.id] ?? 0) > 500) {
+		effectiveTime = reduceNumByPercent(effectiveTime, 15);
+		boosts.push('15% >500 KC');
+	}
+
+	const mage = calcSetupPercent(bisMageGear.stats, user.getGear('mage').stats, 'attack_magic', [
+		'attack_stab',
+		'attack_slash',
+		'attack_crush',
+		'attack_ranged',
+		'defence_crush',
+		'defence_ranged',
+		'defence_slash',
+		'defence_stab'
+	]);
+
+	const melee = calcSetupPercent(bisMeleeGear.stats, user.getGear('melee').stats, 'attack_stab', [
+		'attack_slash',
+		'attack_crush',
+		'attack_ranged',
+		'attack_magic',
+		'defence_magic'
+	]);
+
+	const totalGearPercent = (melee + mage) / 2;
+	const speedReductionMaxPercent = 15;
+	effectiveTime = reduceNumByPercent(
+		effectiveTime,
+		calcPercentOfNum(Math.ceil(totalGearPercent), speedReductionMaxPercent)
+	);
+	boosts.push(
+		`${calcPercentOfNum(Math.ceil(totalGearPercent), speedReductionMaxPercent)}% gear stats [${Math.ceil(
+			melee
+		)}% of max melee, ${Math.ceil(mage)}% of max mage]`
+	);
+
+	for (let itemBoost of itemBoosts) {
+		if (user.getGear(itemBoost.setup).hasEquipped(itemBoost.item.id)) {
+			effectiveTime = reduceNumByPercent(effectiveTime, itemBoost.boost);
+			boosts.push(`${itemBoost.boost}% ${itemBoost.item.name}`);
+		}
+	}
+
+	const maxTripLength = user.maxTripLength('Naxxus');
+	// If no quantity provided, set it to the max.
+	if (quantity === undefined) {
+		quantity = Math.floor(maxTripLength / effectiveTime);
+	}
+	if (quantity * effectiveTime > maxTripLength) {
+		return `The max number of Naxxus you can do is ${Math.floor(maxTripLength / effectiveTime)}!`;
+	}
+
+	const kc = user.settings.get(UserSettings.MonsterScores)[Naxxus.id] ?? 0;
+	let brewsNeeded = 10;
+	if (kc > 500) brewsNeeded *= 0.2;
+	else if (kc > 400) brewsNeeded *= 0.4;
+	else if (kc > 300) brewsNeeded *= 0.6;
+	else if (kc > 200) brewsNeeded *= 0.8;
+
+	brewsNeeded *= quantity;
+
+	const foodBank = new Bank()
+		.add('Enhanced saradomin brew', brewsNeeded)
```

Comment:
While, currently, this is okay, if we ever decided to nerf/buff the brews needed, this will cause you to add a non-integer amount of brews to the cost.